### ISSUE TITLE
bug: U token contract is not evaluated properly

### ISSUE THREAD

**Issue by asiaziola** [warp-contracts/warp-dre-node#45]
The original problem refers to this bazar contract: https://sonar.warp.cc/#/app/contract/XVO04wSSqoyrPN5ut4jtFjKc_c41k6nUhpilohRDilM?network=mainnet

Contract (contract A) has `createOrder` function which writes to two other contracts - token contract (contract B) and asset contract (contract C). Most of the time it seems that dre nodes don't catch either one or two internal writes and `createOrder` interaction is not properly registered in two external contracts. Team responsible for the bazar implementation added a dummy interaction after each `createOrder` interaction so dre would recalculate the state of the contract.

Our idea was to prepare a dedicated dre node which would evaluate only the state of contract A. 

First problem which I encountered was that evaluating the state locally for contract A was non-deterministic. Contract A - in the `handle` function has been reading the state of U token contract which has nearly 90k interactions - https://sonar.warp.cc/#/app/contract/KTzTXT_ANmF84fWEKHzWURD1LWd9QaFR9yfYUwH2Lxw?network=mainnet

Evaluating the state of U token contract resulted in interactions erroring with an error message `timeout`. The default setting for dre nodes has `maxInteractionEvaluationTimeSeconds` set to 10s - when I increased this value, I got deterministic results. However - time of evaluation seems to be quite long.

Running a dedicated node also did not succeed. Dre node has been configured to react only to contract A notifications, however - U token which is read inside contract A has not been evaluated properly (the results are different from other dre nodes so in the end I'm not sure which results are proper :))  

1. `i-2T9huigZODi0XnvN6ZegJwc3Z-F4i3PCsAt4_ppyw` contract is writing to U token contract - it has `useConstructor` option set in the contract manifest but it does not have __init function and therefore it errors. It seems right but - it's weird that contract does not get blacklisted and error is thrown also for U contract - resulting with U contract being blacklisted (e.g. - DRE-6). Surprisingly one of the interactions of contract A - also throws a `useConstructor` error message.
2. I've added `i-2T9huigZODi0XnvN6ZegJwc3Z-F4i3PCsAt4_ppyw` contract to a blacklist and then - based on the ContractBlacklistPlugin implementation - `SkipUnsafeError` has been thrown (also resulting with U token contract blacklisting)
3. I've added `SkipUnsafeError` to an error whitelist but then evaluation has hung after `SkipUnsafeError` being thrown - https://github.com/warp-contracts/warp-dre-node/blob/main/src/workers/updateProcessor.js#L41. The evaluation has been then trapped in the loop each time throwing this ContractError.

To summarize - handling properly error from the contract with `useConstructor` set to true would resolve the issue.

I did not analyze it any further. The team responsible for bazar contract changed the contract A to a different id which internally does not read U token and therefore - the calculation seems to be correct. I've configured DRE-5 to calculate following contract: https://sonar.warp.cc/#/app/contract/aJIOMmqqpZ63V_SsBuSKWRILBYKIH967eFzMPHA8GYo?network=mainnet (I also added test U token contract to the evaluation list and imo for the final version the original U should be calculated as well).

Alex team indicated this contract https://sonar.warp.cc/#/app/contract/pKAUV26rFgG13XwS7oZ1IQ8dDIRcdV9xnC8XEnZ7cfQ?network=mainnet# as the final one (which again - read U state).
